### PR TITLE
Refactor Auth Flow, Add Authorization to Story

### PR DESCRIPTION
**Pull request by HarishTeens**
### Important
1. Storing MongoID inside JWT instead of role information.
2. Permissions map is locally hardcoded and populated in context function
3. ### Implementation of Club wise permission check
the permissions string now would be in the pattern of `<permission related to club>$<club ID>`
Having the permissions this way makes is real easy when checking for permission at the resolver level.

### This PR includes
- remove firebase cloudfunctions as it is not needed anymore
- create helpers directory which has all independent functions
- upgrade role on NITR verification
- a user needs to have level 3 access associated with the club, to add story for the same club

### CODE REVIEW

File: sources/resolvers/story.js [hunk 1]
```diff
@@ -14,17 +14,17 @@ const queries = {
...
 const mutations = {
 	addStory: async (parent, { story }, { dataSources ,uid,permissions}, info) => {
-		if (permissions.find((permission) => permission == 'stories.add')) {			
+		if (permissions.find((permission) => permission == 'stories.add$'+story.author)) {			
 			return dataSources.StoryAPI.addStory(story);
 		} else {
-			return [ERRORS.PERMISSION_DENIED];
+			return ERRORS.PERMISSION_DENIED;
 		}		
 	}
```
Comment (HarishTeens) [inline sources/resolvers/story.js:29]: As an example, This resolver expects a string like `stories.add$<club ID>`

File: sources/helpers/permissions.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    foundUser.clubAccess.forEach(accessObj => {
+        if(accessObj.club)
+            rolesSet.add(accessObj.level+"$"+accessObj.club)
+        else
+            rolesSet.add(accessObj.level);  
+    }); 
+    let permissions=[]
+    for(let role of rolesSet){   
+        const splits=role.split("$")     ;
+        console.log(splits);
```
Comment (DesignrKnight) [inline sources/helpers/permissions.js:21]: Is this console.log needed?

File: sources/helpers/firebase.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+const admin = require('firebase-admin');
```
Comment (DesignrKnight) [inline sources/helpers/firebase.js:1]: I like the logic/flow in this file. Awesome actually. Works neatly.

Hopefully🙏

**Comment by DesignrKnight** [comment]
Had caused conflict as I did a PR merge #54 #53 
Synced and resolved. Ran docs CLI also

File: sources/helpers/firebase.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+const serviceAccount = require('../../project-elaichi.json');
+
+// Firebase Init
+const firebaseAPP=admin.initializeApp({
```
[the remark this answers is not in the document]
Comment (HarishTeens) [reply sources/helpers/firebase.js:5]: Isnt it firebaseAPP everywhere?

[the remark this answers is not in the document]
Comment (DesignrKnight) [reply sources/helpers/firebase.js:5]: I meant consistent style across the project. All caps seem weird halfway in variable name

**Comment by DesignrKnight** [comment]
Is there anything pending on this?

File: sources/helpers/permissions.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    "2":["users.secretEvents"],
+    "3":["stories.add"]
+}
+
+const populatePermissions=async (id)=>{
+    const foundUser= await Users.findById(id).populate("clubAccess");
+    let rolesSet=new Set();
+    foundUser.clubAccess.forEach(accessObj => {
+        if(accessObj.club)
+            rolesSet.add(accessObj.level+"$"+accessObj.club)
```
Comment (roshankshaw) [inline sources/helpers/permissions.js:14]: The encoding scheme isn't required, we can use an object with key value pairs.

**Review by roshankshaw — COMMENTED** [review]
Docs changes should not be included in the PR

**Comment by DesignrKnight** [comment]
> Docs changes should not be included in the PR

Should I remove that?
I can do

File: sources/helpers/permissions.js (hunk 5 above)
Comment (HarishTeens) [reply sources/helpers/permissions.js:14]: i will merge now, encoding algorithm could be changed later

**Pull request merged**